kratos/sources/constitutive_law.cpp

```diff
@@ -688,7 +692,7 @@ void ConstitutiveLaw::InitializeMaterialResponsePK2 (Parameters& rValues)
  */
 void ConstitutiveLaw::InitializeMaterialResponseKirchhoff (Parameters& rValues)
 {
-    KRATOS_ERROR << "Calling virtual function for InitializeMaterialResponseKirchhoff" << std::endl;
+    KRATOS_ERROR_IF(RequiresInitializeMaterialResponse()) <<  "Calling virtual function for InitializeMaterialResponseKirchhoff. Please implement InitializeMaterialResponseKirchhoff or RequiresInitializeMaterialResponse in case this CL does not require it" << std::endl;

```

Comment:
```suggestion
    KRATOS_ERROR_IF(this->RequiresInitializeMaterialResponse()) <<  "Calling virtual function for InitializeMaterialResponseKirchhoff. Please implement InitializeMaterialResponseKirchhoff or RequiresInitializeMaterialResponse in case this CL does not require it" << std::endl;
```